src/GraphWebhooks/Services/CertificateService.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Azure.Identity;
+using Azure.Security.KeyVault.Certificates;
+using Azure.Security.KeyVault.Secrets;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace GraphWebhooks.Services
+{
+    /// <summary>
+    /// Implements methods to retrieve certificates from Azure Key Vault
+    /// </summary>
+    public class CertificateService
+    {
+        private readonly IConfiguration _config;
+        private readonly ILogger<CertificateService> _logger;
+        private byte[] _publicKeyBytes = null;
+        private byte[] _privateKeyBytes = null;
+
+        public CertificateService(
+            IConfiguration configuration,
+            ILogger<CertificateService> logger)
+        {
+            _config = configuration;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Gets the configured public key from the Azure Key Vault
+        /// </summary>
+        /// <returns>The public key</returns>
+        public async Task<X509Certificate2> GetEncryptionCertificate()
+        {
+            if (_publicKeyBytes == null)
+            {
+                await LoadCertificates();
+            }
+
+            return new X509Certificate2(_publicKeyBytes);
```

Comment:
as those are disposable, we should probably get a hold of those instead of instantiating it every time from the bytes

Reply:
Well actually, this was causing an issue. I originally was saving the X509Certificate2 objects as private variables, but the private key became invalid after the first use. I verified that the service (which is added as a singleton in Startup.cs) was staying in memory, and the variable wasn't null, but all of it's members were invalid, and using it gave a "m_safecertcontext is an invalid handle" error.

Reply:
alright then, let's make sure we dispose the value in the caller after use.

Reply:
Yep, Graph SDK wraps it in a `using`, and I believe the error I'm seeing there is because it's getting disposed :D